### PR TITLE
Menu: Extend Scrolling demos for JS frameworks

### PR DESCRIPTION
**Pull request by jdvictoria**

### CODE REVIEW

**Comment by jdvictoria** [comment]
[Menu feature: extend the Scrolling demo for frameworks](https://trello.com/c/2x6raccH/8782-menu-feature-extend-the-scrolling-demo-for-frameworks)

**Review by iBat — DISMISSED** [review]

File: apps/demos/Demos/Menu/Scrolling/Angular/app/app.component.css [hunk 1]
```diff
@@ -0,0 +1,38 @@
+html,
+body {
+  width: 100%;
+  height: 100%;
+}
+
+.demo-container {
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/Menu/Scrolling/Angular/app/app.component.css]: we use ::ng-deep for angular styles

File: apps/demos/Demos/Menu/Scrolling/React/App.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    if (!e.itemData.items) {
+      notify(`The "${e.itemData.text}" item was clicked`, 'success', 1500);
+    }
+  }, []);
+
+  const limitSubmenuHeightOnMouseClick = useCallback((e: CheckBoxTypes.ValueChangedEvent) => {
+    setLimitSubmenuHeight(e.value);
+  }, [setLimitSubmenuHeight]);
+
+  const onSubmenuShowing = useCallback(({ submenuContainer }: any) => {
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/Menu/Scrolling/React/App.tsx]: not sure, probably this is of type HTMLElement, plz check if we can get rid of any

Comment (AlexanderMoiseev) [reply apps/demos/Demos/Menu/Scrolling/React/App.tsx]: i meant by specifying HTMLElement type, not by removing it )

File: apps/demos/Demos/Menu/Scrolling/Angular/app/app.component.css [hunk 3]
```diff
@@ -0,0 +1,38 @@
+html,
+body {
+  width: 100%;
+  height: 100%;
+}
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/Menu/Scrolling/Angular/app/app.component.css]: do we need these styles? isn't it supposed to work the same without it?

Comment (jdvictoria) [reply apps/demos/Demos/Menu/Scrolling/Angular/app/app.component.css]: Without those, Angular TestCafe fails. I applied how it was styled initially on jQuery which was approved last week. It's possible to not have these but we need to remove it for all frameworks.

File: apps/demos/Demos/Menu/Scrolling/Vue/App.vue [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+const products = ref(service.getProducts());
+const limitSubmenuHeight = ref(false);
+
+function itemClick(e: any) {
+  if (!e.itemData.items) {
+    notify(`The "${e.itemData.text}" item was clicked`, 'success', 1500);
+  }
+}
+
+function onSubmenuShowing({ submenuContainer }: any) {
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/Menu/Scrolling/Vue/App.vue]:
```suggestion
function onSubmenuShowing({ submenuContainer }: HTMLDivElement) {
```
or HTMLElement, dunno what kind of element is this

File: apps/demos/Demos/Menu/Scrolling/Angular/app/app.component.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  SUBMENU_HEIGHT = 200;
+
+  limitSubmenuHeight = false;
+
+  constructor(service: Service) {
+    this.products = service.getProducts();
+  }
+
+  itemClick(e: any) {
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/Menu/Scrolling/Angular/app/app.component.ts]: probably this is of type ItemClickEvent or something

**Review by AlexanderMoiseev — DISMISSED** [review]

**Review by iBat — APPROVED** [review]

**Review by AlexanderMoiseev — APPROVED** [review]

**Pull request merged**